numactl: add page (#5167)
tldr-pages/tldr: pages/linux/numactl.md

# numactl

> Control NUMA policy for processes or shared memory.

> 

Reviewer: Please add a more information link:
```suggestion
> More information: <https://man7.org/linux/man-pages/man8/numactl.8.html>.

```

- Run command on node 0 with memory allocated on node 0 and 1:

> `numactl --cpunodebind=0 --membind=0,1 {{command}}`

Reviewer: How can I pass arguments to the command?
```suggestion
`numactl --cpunodebind={{0}} --membind={{0,1}} {{command}} {{command_arguments}}`
```

Reviewer 2: @navarroaxel I updated your suggestion to use the advisable syntax for command arguments (`--`)

- Run command on CPUs (cores) 0-4 and 8-12 of the current cpu-set:

> `numactl --physcpubind=+0-4,8-12 {{command}}`

Reviewer: ```suggestion
`numactl --physcpubind={{+0-4,8-12}} {{command}} {{command_arguments}}`
```

- Run command with its memory interleaved on all CPUs:

> `numactl --interleave=all {{command}}`

Reviewer: The token syntax for argument option:
```suggestion
`numactl --interleave={{all}} {{command}} {{command_arguments}}`
```

Reviewer 2: @navarroaxel I updated your suggestion to use the advisable syntax for command arguments (`--`)

# numactl

> Control NUMA policy for processes or shared memory.
> More information: <https://man7.org/linux/man-pages/man8/numactl.8.html>.

> - Run command on node 0 with memory allocated on node 0 and 1:

Reviewer 2: ```suggestion
- Run a command on node 0 with memory allocated on node 0 and 1:
```

`numactl --cpunodebind={{0}} --membind={{0,1}} -- {{command}} {{command_arguments}}`

> - Run command on CPUs (cores) 0-4 and 8-12 of the current cpu-set:

Reviewer 2: ```suggestion
- Run a command on CPUs (cores) 0-4 and 8-12 of the current cpu-set:
```

> - Run a command on CPUs (cores) 0-4 and 8-12 of the current cpu-set:

Reviewer 2: The man page says `cpuset` everywhere...
```suggestion
- Run a command on CPUs (cores) 0-4 and 8-12 of the current cpuset:
```

Author: Yup, cpuset is correct: https://man7.org/linux/man-pages/man7/cpuset.7.html

> `numactl --physcpubind={{+0-4,8-12}} {{command}} {{command_arguments}}`

Reviewer 2: ```suggestion
`numactl --physcpubind={{+0-4,8-12}} -- {{command}} {{command_arguments}}`
```

Author: Have you ever had issues without the --? If so, where?

Reviewer 2: I haven't ever even used this command :D But I think it's always best to have the `--` because a great many commands run subcommands like this and you can use `--` with them.  It's also more verbose, which I think would be useful to new users of the command (like me for example :)

Reviewer 2: IMO it also makes it quite a bit clearer what args are for `numactl` and what are for the command you're running with numactl.

Author: I understand! Thank you for taking some time to explain your thoughts on the subject; greatly appreciated :D

> - Run command with its memory interleaved on all CPUs:

Reviewer 2: ```suggestion
- Run a command with its memory interleaved on all CPUs:
```

Author: I understand, I was referencing the {{command}} with that sentence. So like saying: run mkdir with ...
I feel I still need to get a hold of the correct wording for tldr :D
